Assets/Scripts/ProceduralGeneration.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System.Collections;
+using UnityEngine;
+using System.Collections.Generic;
+using Quaternion = UnityEngine.Quaternion;
+using Random = UnityEngine.Random;
+using Vector3 = UnityEngine.Vector3;
+
+
+public class ProceduralGeneration : MonoBehaviour
+{
+    // Unit cubes should evenly subdivide chunk, and chunks should evenly subdivide level.
+    public Vector3Int levelDimensions;
+    public int chunkSize;
+    public int unitCubeSize;
+    public float asteroidDensity = 0.5f;
+    public Vector2 asteroidSizeRange;
+    public float viewDistance = 500f;
+    
+    // public GameObject blackHolePrefab;
+    // public GameObject exitPortalPrefab;
+    public GameObject asteroidPrefab;
+
+    private GameObject asteroidsCollection;
+    
+    private Vector3Int numChunksInLevel;
+    private Vector3Int numUnitCubesInChunk;
+
+    private CullingGroup cullGroup;
+    private BoundingSphere[] bounds;
+    private Dictionary<int, GameObject> entityLookup;
+    private int EntityCount => entityLookup.Count;
+
+    private Dictionary<int, Transform> entityTransforms;
+    
+    private Dictionary<int, int> entityIdToIndex;
+    private Dictionary<int, int> entityIndexToId;
```

Comment:
Can you rename these to something like `entityIdToSphereIndex` and `sphereIndexToEntityId` for clarity?

Reply:
sphere doesn't mean anything extra, but it's basically the entity index across all entities...

Reply:
It's referring to the index of the bounding sphere though